tests/core/versions/testversion1/BatchWriter.ts

```diff
@@ -0,0 +1,18 @@
+import IBatchWriter from '../../../../lib/core/interfaces/IBatchWriter';

```

Comment:
Nit: Consider following more of javascript convension of using hyphens between words.
`testversion1` -> `test-version-1` or maybe avoiding it altogether to `0.0.1`

Reply:
Renaming.